.editorconfig

```diff
@@ -0,0 +1,16 @@
+# https://EditorConfig.org
+root = true
+
+[*]
+charset = utf-8
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+indent_style = tab
+tab_width = 4
+indent_size = 4
```

Comment:
Missing directive:

```
end_of_line = lf

```

Reply:
I left that one intentionally unset, because it would conflict with git's `autocrlf` feature on Windows. If we wanted `lf` line endings everywhere we would need to reconfigure git via `.gitattributes`. However, this will cause conflicts with existing clones on windows.

Reply:
Hm yes, the `.editorconfig` controls what the editor does in the user's local environment, and if that environment is Windows, then I can imagine that the user wants "the usual" line terminators. I can respect not messing with the user's preference in their own environment.

However, *for commits to this project* I would like to standardize on `\n` as the line ending. I guess the `.editorconfig` might not be the best place to encode that, and a better place would indeed be `.gitattributes`, or perhaps some sort of style check in CI.

Reply:
But isn't the `.editorconfig` file intended as a hint to the editor on how to save files that are intended to be *commits for this project*? In that case I think setting the end-of-line preference might be fair after all.

I think it shouldn't conflict with `autocrlf` because when Git checks out the files, it will change `\n` to `\r\n`, but then the editor will map them back to `\n`.